### ISSUE TITLE
[Feature request] Multimodal support

### ISSUE THREAD

**Issue by evilsocket** [dreadnode/rigging#7]
I've read the docs and rigging doesn't seem to currently support multimodal LLMs, do you have any plan for that? It'd be interesting to test llava-phi3 or llava-llama3 with text + image input (i'm thinking the screenshot of a webpage / terminal / etc).

**Comment by monoxgas** [comment]
We do have plans to add support for those additional formats, it seems like the OpenAI spec already has good adoption for things like ImageParts on messages. We'll look at getting this added in a future version!

**Comment by evilsocket** [comment]
thanks @monoxgas, love the project, looking forward to it

**Comment by monoxgas** [comment]
Early support for this is currently in the dev branch: https://github.com/dreadnode/rigging/commit/25af782fe71dea3df8d78d90cb4fb055646aa104

```py
import rigging as rg

image = rg.ContentImageUrl.from_file('../docs/assets/rigging.png')
message = rg.Message("user", ["What'set in this image?", image])

chat = (
    await
    rg.get_generator("vertex_ai/gemini-1.5-pro")
    .chat(message)
    .run()
)

print(chat.last)
```

Or the more verbose JSON format:
```py
import rigging as rg

chat = (
    await
    rg.get_generator("vertex_ai/gemini-1.5-pro")
    .chat(
        {
            "role": "user",
            "content": [
                {"type": "text", "text": "Whats in this image?"},
                {
                    "type": "image_url",
                    "image_url": {
                        "url": "https://upload.wikimedia.org/wikipedia/commons/thumb/d/dd/Gfp-wisconsin-madison-the-nature-boardwalk.jpg/2560px-Gfp-wisconsin-madison-the-nature-boardwalk.jpg"
                    },
                },
            ],
        }
    )
    .run()
)
print(chat.last)
```

**Comment by monoxgas** [comment]
Released in [v2.0.2](https://github.com/dreadnode/rigging/releases/tag/v2.0.2)
